src/pipdeptree/_models/dag.py

```diff
@@ -36,19 +36,14 @@ class PackageDAG(Mapping[DistPackage, List[ReqPackage]]):
     """
 
     @classmethod
-    def from_pkgs(cls, pkgs: list[DistInfoDistribution]) -> PackageDAG:
-        dist_pkgs = [DistPackage(p) for p in pkgs]
+    def from_pkgs(cls, pkgs: list[Distribution]) -> PackageDAG:
+        dist_pkgs = [DistPackage(p) for p in pkgs if p.metadata["Name"]]
```

Comment:
As we discussed [here](https://github.com/tox-dev/pipdeptree/pull/333#discussion_r1527659369), we can get rid of the `if p.metadata["Name"]` check here.

It do feel like it will become a problem in the future, but we'll figure out a way to handle if we need to.

Reply:
fixed.